src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8026,22 +8045,22 @@ internal static bool IsReparsePoint(FileSystemInfo fileInfo)
             return fileInfo.Attributes.HasFlag(System.IO.FileAttributes.ReparsePoint);
         }
 
-        internal static bool IsNameSurrogateReparsePoint(string filePath)
+        internal static bool IsReparsePointWithTarget(FileSystemInfo fileInfo)
         {
+            if (!IsReparsePoint(fileInfo))
+            {
+                return false;
+            }
 #if !UNIX
             var data = new WIN32_FIND_DATA();
-            using (SafeFileHandle handle = FindFirstFileEx(filePath, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
+            using (SafeFileHandle handle = FindFirstFileEx(fileInfo.FullName, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
             {
-                // Name surrogates are reparse points that point to other named entities local to the filesystem (like symlinks)
-                // In the case of OneDrive, they are not surrogates and would be safe to recurse into.
-                // This code is equivalent to the IsReparseTagNameSurrogate macro: https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/ntifs/nf-ntifs-isreparsetagnamesurrogate
```

Comment:
Why do we remove the comment? Without it the code is a riddle.

Reply:
Yeah, I guess no reason it shouldn't remain.  I'll add it back.